pkg/quota/evaluator/core/pods.go

```diff
@@ -255,17 +257,19 @@ func PodUsageFunc(obj runtime.Object) (api.ResourceList, error) {
 	if err != nil {
 		return api.ResourceList{}, err
 	}
-	// Only charge pod count for uninitialized pod.
-	if !initialization.IsInitialized(pod.Initializers) {
-		result := api.ResourceList{}
-		result[api.ResourcePods] = resource.MustParse("1")
-		return result, nil
-	}
 
 	// by convention, we do not quota pods that have reached an end-of-life state
 	if !QuotaPod(pod) {
 		return api.ResourceList{}, nil
 	}
+	// Only charge pod count for uninitialized pod.
+	if utilfeature.DefaultFeatureGate.Enabled(features.Initializers) {
```

Comment:
Perhaps the BeforeUpdate/BeforeCreate should fail the request if feature gate is disabled and initializers are set, instead of repairing the request. Then we don't have to deal with feature gate here. Admission is called before BeforeUpdate/BeforeCreate, it's hard to reasoning the correct admission behavior if the object might get changed later.

Reply:
Feature gate off should behave as if those fields don't exist. We could drop pre-admission, I suppose